### PR TITLE
Implement Account Switch Key ( ASK ) and refactor structs

### PR DESCRIPTION
**Pull request by RafPe**
* Implements capability to use ASK 
* Implements new API call method
* Implements structured way for query string params 
* Implements support to add additional headers in request
* Adds additional debug info by dumping response 
* Updates logrus dependency to 1.2
* Removes Akamai prefix from structs
* Adds const string types

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -357,3 +380,153 @@ func prepareURL(url *url.URL, path string) (*url.URL, error) {
...
+
+	return v.Encode(), nil
+}
+
+// makeAPIRequest creates an HTTP request that can be sent to Akamai APIs. It will handle security headers and signinig of the request.
+//
+func (cl *Client) makeAPIRequest(method, path string, queryParams, structResponse, structRequest interface{}, headers map[string]string) (*ClientResponse, error) {
+
+	log.Debug("[NewRequest]::Prepare URL for http request")
+	targetURL, _ := prepareURL(cl.baseURL, path)
```
Comment (partamonov) [inline client.go]: Do not ignore error, handle it

File: client.go [hunk 2]
```diff
@@ -343,9 +362,13 @@ func (cl *Client) SetBaseURL(urlStr string, passThrough bool) error {
 	return err
 }
 
+// EnableASK instructs client to use ASK
+func (cl *Client) EnableASK() {
+	log.Debug("[EnableASK]::Enabling ASK ...")
```
Comment (partamonov) [inline client.go]: decode the message to Enable Account Switching Key

File: client.go [hunk 3]
```diff
@@ -44,18 +43,30 @@ const (
...
 
 const (
 	Production AkamaiEnvironment = "production"
 	Staging    AkamaiEnvironment = "staging"
 )
 
+// AkamaiRequestFrom represents Akamai's source for request.
+type AkamaiRequestFrom string
+
+const (
```
Comment (partamonov) [inline client.go]: Let's add constants on top of the file and in one place

File: client.go [hunk 4]
```diff
@@ -343,9 +362,13 @@ func (cl *Client) SetBaseURL(urlStr string, passThrough bool) error {
...
+	log.Debug("[EnableASK]::Enabling ASK ...")
+	cl.accountSwitchEnabled = true
+}
+
 // prepareURL returns URL which is used to make API call
-//
-// client
 func prepareURL(url *url.URL, path string) (*url.URL, error) {
 
 	rel, err := url.Parse(strings.TrimPrefix(path, "/"))
```
Comment (partamonov) [inline client.go:370]: error is not returned if appeared

File: client.go [hunk 5]
```diff
@@ -357,3 +367,153 @@ func prepareURL(url *url.URL, path string) (*url.URL, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	/*
+		Modify request for POST/PUT
+	*/
+	if method == http.MethodPost || method == http.MethodPut {
+		log.Info("Prepare request body object")
+		log.Debug("[NewRequest]::Method is POST/PUT")
```
Comment (partamonov) [inline client.go]: Why not specify exact method instead of PUT/POST guess

File: client.go (hunk 3 above)
Comment (RafPe) [reply client.go]: Already fixed https://github.com/apiheat/go-edgegrid/blob/23abfca561129b7d5ffb15dbd6ca01c388f5effc/client.go

File: client.go [hunk 6]
```diff
@@ -357,3 +367,153 @@ func prepareURL(url *url.URL, path string) (*url.URL, error) {
...
+
+		if queryStringError != nil {
+			log.Debug("[NewRequest]::Error adding query string parameters")
+			log.Debug(fmt.Sprintf("[NewRequest]:: %s", queryStringError.Error()))
+			return nil, queryStringError
+		}
+	}
+
+	/*
+		Add signature header
```
Comment (partamonov) [inline client.go:467]: Why not to move signature header closer to previous header block to have headers in same place and near

File: client.go (hunk 5 above)
Comment (RafPe) [reply client.go]: https://github.com/apiheat/go-edgegrid/pull/21/commits/9dac2f34486a2acc758701b882162d3aa6196871

File: client.go [hunk 7]
```diff
@@ -357,3 +367,153 @@ func prepareURL(url *url.URL, path string) (*url.URL, error) {
...
+		log.Debug("[NewRequest]::Error reading response body")
+		log.Debug(fmt.Sprintf("[NewRequest]:: %s", err.Error()))
+		return nil, err
+	}
+
+	log.Debug("[NewRequest]::Set client object response and body")
+	clientResp.Response = resp
+	clientResp.Body = string(byt)
+
+	log.Debug("[NewRequest]::Response code is:" + strconv.Itoa(resp.StatusCode))
```
Comment (partamonov) [inline client.go]: Both status code and body was already given in debug log - with respDump

File: client.go (hunk 6 above)
Comment (RafPe) [reply client.go:467]: Fixed

File: client.go (hunk 7 above)
Comment (RafPe) [reply client.go]: Fixed

File: client.go (hunk 2 above)
Comment (RafPe) [reply client.go]: Done https://github.com/apiheat/go-edgegrid/pull/21/commits/523bea08d28440494671e09ef94132c42962c848

File: client.go (hunk 1 above)
Comment (RafPe) [reply client.go]: Fixed https://github.com/apiheat/go-edgegrid/pull/21/commits/97f4a8d8dc4cd13cf194c3a300ccff67fd8d7045

**Pull request merged**